ui/src/features/stage/create-stage.tsx

```diff
@@ -148,12 +160,32 @@ export const CreateStage = ({
       <Tabs
         onChange={(newTab) => {
           if (tab === 'wizard' && newTab === 'yaml') {
-            setValue('value', stageFormToYAML(watch(), project || ''));
+            setValue(
+              'value',
+              stageFormToYAML(watch(), project || '', promotionWizardStepsState.getYAML())
+            );
+          } else {
+            const yaml = getValues('value');
+
+            try {
+              const stageSpec: Stage = parse(yaml);
+
+              promotionWizardStepsState.setYAML(
+                stringify(stageSpec?.spec?.promotionTemplate?.spec?.steps)
+              );
+            } catch (e) {
+              // explicitly ignore
```

Comment:
Don't we want to inform the user that yaml is broken?

Reply:
This is tricky. If we block user navigating if YAML is invalid, this becomes too aggressive. If we unblock user navigating and show error if YAML is invalid, we already lost broken YAML when we navigate. If we want to keep the broken YAML, this becomes complicated of syncing 2 source of truths (local state and YAML state).

So to simplify at this point, I think the use-case will either be complete raw YAML or complete form or correct YAML to wizard transformation. Broken YAML is rare case I think